src/replica/replica.cpp

```diff
@@ -176,6 +176,19 @@ void replica::on_client_read(dsn::message_ex *request, bool ignore_throttling)
         response_client_read(request, ERR_ACL_DENY);
     }
 
+    // validate partition_hash
```

Comment:
I think you can implement a func, and move these code into it

Reply:
I meant to add another function to do it, but it can only be implemented as macro, read request should call `on_client_read`, write request should call `on_client_write`. Besides, I think it clear to leave return error in `on_client_read` and `on_client_write`.